report/generate_trustymail_report.py

```diff
@@ -438,7 +439,15 @@ def __score_domain(self, domain):
 
             score['dmarc_subdomain_policy'] = domain['dmarc_subdomain_policy']
             score['valid_dmarc_subdomain_policy_reject'] = False
-            if (
+            # According to RFC7489, "'sp' will be ignored for DMARC
+            # records published on subdomains of Organizational
+            # Domains due to the effect of the DMARC policy discovery
+            # mechanism."  Therefore we have chosen not to penalize
+            # for sp!=reject when considering subdomains.
+            #
+            # See here for more details:
+            # https://tools.ietf.org/html/rfc7489#section-6.3
+            if not domain['is_base_domain'] or (
```

Comment:
Doesn't this logic allow a subdomain with invalid DMARC to get `valid_dmarc_subdomain_policy_reject = True`?  Maybe that's ok?  I'm not quite sure.

Reply:
I think you're right.  This should be:
```python
if score['valid_dmarc'] and (not domain['is_base_domain'] or domain['dmarc_subdomain_policy'] == 'reject'):
```

Does that look correct to you?

Reply:
Yes, I think that is better.  Don't forget to make the corresponding changes in the Cyber Exposure scorecard logic.

Reply:
Good catch!
:eagle: :eye:

Reply:
Fixed in 6ad8b62.